pkg/pillar/cmd/zedagent/localinfo.go

```diff
@@ -137,17 +135,126 @@ func sendLocalAppInfo(ctx *getconfigContext) {
 				//throttle sending to be about once per hour
 				updateLocalAppInfoTicker(ctx, true)
 			}
-			if resp.StatusCode != http.StatusOK {
+			if resp.StatusCode != http.StatusOK && resp.StatusCode != http.StatusNoContent {
 				errList = append(errList, fmt.Sprintf("SendLocal: wrong response status code: %d",
 					resp.StatusCode))
 				continue
 			}
+			if resp.StatusCode == http.StatusNoContent {
+				log.Functionf("Local server %s does not require additional app commands to execute",
+					localServerURL)
+				return nil
+			}
+			if appCmds.GetServerToken() != ctx.profileServerToken {
+				errList = append(errList,
+					fmt.Sprintf("invalid token submitted by local server (%s)", appCmds.GetServerToken()))
+				continue
+			}
 			updateLocalAppInfoTicker(ctx, false)
-			lastLocalAppInfoSentHash = newHash
-			return
+			return appCmds
 		}
 	}
 	log.Errorf("sendLocalAppInfo: all attempts failed: %s", strings.Join(errList, ";"))
+	return nil
+}
+
+func processReceivedAppCommands(ctx *getconfigContext, cmdList *profile.LocalAppCmdList) {
+	ctx.localAppCommandsLock.Lock()
+	defer ctx.localAppCommandsLock.Unlock()
+	if cmdList == nil {
+		// Nothing requested by local server, just refresh the persisted config.
+		touchAppCommands()
```

Comment:
This raises the question whether you are using the same max lifetime for the local app commands as for the persisted config from the controller.
(Separately one could argue that even the latter should be infinite but for the app commands wouldn't inifiite be the best?)

Reply:
I think that after 10 minutes the user will loose patience anyway and will click the "purge" or "restart" button again.

Reply:
So shouldn't we make the default timer value be 30 seconds or less?